src/JPEGStream.h

```diff
@@ -23,18 +23,15 @@ init_closure_destination(j_compress_ptr cinfo){
 
 boolean
 empty_closure_output_buffer(j_compress_ptr cinfo){
-  Nan::HandleScope scope;
-  Nan::AsyncResource async("canvas:empty_closure_output_buffer");
   closure_destination_mgr *dest = (closure_destination_mgr *) cinfo->dest;
+  Napi::Env env = dest->closure->canvas->Env();
+  Napi::HandleScope scope(env);
+  Napi::AsyncContext async(env, "canvas:empty_closure_output_buffer");
 
-  v8::Local<v8::Object> buf = Nan::NewBuffer((char *)dest->buffer, dest->bufsize).ToLocalChecked();
+  Napi::Object buf = Napi::Buffer<char>::New(env, (char *)dest->buffer, dest->bufsize);
```

Comment:
Wrapping an existing pointer in a nodejs Buffer are not compatible with recent versions of electron.
See https://github.com/nodejs/node-addon-api/blob/main/doc/external_buffer.md
It looks like you might have some other occurrences of this too

Reply:
Thanks for pointing that out. So I guess we can't claim Electron 21+ support yet. I'm going to consider that out of scope for this PR since we used external memory before these changes, but we should support Electron too without impacting Node performance. I wonder if Node will ever make the same change that Electron did too...

I'm so curious to see the use case for node-canvas in Electron. I thought you would just use `<canvas>`...

Reply:
> but we should support Electron too without impacting Node performance.

There is an alternate function `Napi::Buffer<char>::NewOrCopy(` designed for this case. Depending on whether you use `dest` again after this call, it may not require any more changes than that.

> I'm so curious to see the use case for node-canvas in Electron. I thought you would just use <canvas>...

That would depend on whether you want to use it on the backend or frontend of the application, but I agree it does sound like an usual case.

I have a couple of applications which use electron for packaging and to get a tray icon, nothing more. They are also possible to run in plain nodejs. It might technically be possible to use a canvas element, but it wouldnt be nice.
I found [a library for a usb device](https://github.com/foxxyz/loupedeck) which uses canvas for generating the images to display. I think that one is quite likely to be in electron instead of nodejs

Reply:
Actually there is a HUGE use case in electron. It is why myself and many have been praying for node-canvas adoption of N-API ;) as mentioned above a big use case is because in electron you split everything into renderer and main threads. In main, where you do all your node stuff, you cannot use <canvas> and hence the need for node-canvas. Passing stuff between main and renderer is a pain (via IPC) so it is very useful to have node-canvas in main thread.

Lastly, and most important for me, Mozilla pdf-js relies on node-canvas and I use it for many things in my electron app. Without node-canvas napi compatibility I am not able to use pdf-js in newer versions of electron.

Does that make sense? Any questions?

Reply:
Yeah we are also relying on node-canvas to handle rendering the contents of the <canvas> on the renderer side as a PDF or SVG in the main process. Node-canvas support for rendering to PDF/SVG is a great plus in this case, and would not be possible to do in the main thread.

Hoping that this would work with Electron versions 21+, but honestly if this already works with Electron version 20, that's a huge improvement already, so thank you for your work!